### PR TITLE
test: Improve use of test fixtures in integration tests

### PR DESCRIPTION
**Pull request by maybe-sybr**
## Description
Test suite improvements based on the time I spent in the canvas tests for #6770 . This
changeset is based on that PR but could be separated if necessary - TBH I think it's
easiest to just block merge of this on that one.

### CODE REVIEW

**Comment by maybe-sybr** [comment]
Marking this as a draft to ensure we don't merge it prior to #6770 going in.

**Comment by maybe-sybr** [comment]
I've merged in the ci branch from #6649 to see if it runs these tests happily or not.

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -71,3 +72,71 @@ jobs:
             flags: unittests # optional
             fail_ci_if_error: true # optional (default = false)
             verbose: true # optional (default = false)
+
+
+  Integration:
```
Comment (auvipy) [inline .github/workflows/python-package.yml:77]: this might need some adjustment, I will try to push them this week After  my Eid vacation ends

Comment (maybe-sybr) [reply .github/workflows/python-package.yml:77]: No rush, this won't go anywhere until after 5.1 anyway

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -71,3 +72,71 @@ jobs:
...
+        env:
+        # The hostname used to communicate with the Redis service container
+          REDIS_HOST: localhost
+          # The default Redis port
+          REDIS_PORT: 6379
+
+    steps:
+    - name: Install apt packages
+      run: |
+        sudo apt-get install -f libcurl4-openssl-dev libssl-dev gnutls-dev httping expect libmemcached-dev
```
Comment (thedrow) [inline .github/workflows/python-package.yml:114]:
```suggestion
        sudo apt update && sudo apt install -f libcurl4-openssl-dev libssl-dev gnutls-dev httping expect libmemcached-dev
```

File: requirements/test-ci-base.txt [hunk 3]
```diff
@@ -1,5 +1,4 @@
 pytest-cov
-pytest-travis-fold
```
Comment (thedrow) [inline requirements/test-ci-base.txt:2]: Good idea.

**Comment by maybe-sybr** [comment]
@thedrow - those diffs are from #6649 rather than my own changes. Probably best to transplant those comments over there for @auvipy / whoever picks that works up. My diff is just 1095f05...dfae402. Don't stress too much about this PR for the moment, I'll re-request review when it's cleaner and has a working base for integration tests in CI.

**Comment by maybe-sybr** [comment]
This has been rebased on top of current master but still merges in an old version of the `ci` branch up for merge in #6649

**Pull request closed without merging**